modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogTableDescriptor.java

```diff
@@ -282,4 +250,199 @@ public String storageProfile() {
         return storageProfile;
     }
 
+    /**
+     * {@code CatalogTableDescriptor} builder static inner class.
+     */
+    public static final class Builder {
+        private int id;
+        private String name;
+        private int zoneId;
+        private int schemaId;
+        private int pkIndexId;
+        private CatalogTableSchemaVersions schemaVersions;
+        private List<CatalogTableColumnDescriptor> columns;
+        private List<String> primaryKeyColumns;
+        @Nullable private List<String> colocationColumns;
+        private String storageProfile;
+        private HybridTimestamp timestamp = INITIAL_TIMESTAMP;
+        private int tableVersion = 0;
```

Comment:
why do we need explicit field for `tableVersion`? It's always been representing version of the table row, not the table itself.